synapse/storage/databases/main/room.py

```diff
@@ -1274,6 +1274,29 @@ async def is_partial_state_room(self, room_id: str) -> bool:
 
         return entry is not None
 
+    @cachedList(cached_method_name="is_partial_state_room", list_name="room_ids")
+    async def is_partial_state_rooms(
```

Comment:
"are partial state rooms" maybe? That still doesn't read well though.

```suggestion
    async def are_partial_state_rooms(
```

Reply:
As I mentioned below I really want this to be `get_partial_state_rooms` which only returns partial stated rooms...

Reply:
I agree. I like that having the same prefix helps with tab complete, so if you remember `Is_partial<tab>` should get you both

Reply:
`is_partial_state_room_batch` or `_batched`? maybe

Reply:
I've done this in e01b43ad78.